Add API filtering documentation in 5.6 (#1407)
sensu/sensu-docs: content/sensu-go/5.6/api/overview.md

]
{{< /highlight >}}

### Filtering

**ENTERPRISE ONLY**: API filtering in Sensu Go requires a Sensu Enterprise license. To activate your Sensu Enterprise license, see the [getting started guide][7].

The Sensu API supports filtering for all GET endpoints that return an array. You can filter resources based on their labels with a label selector using the `labelSelector` query parameter and on certain pre-determined fields with a field selector using the `fieldSelector` query parameter.

> For example, the following request filters the response to only include resources that have a label entry `region` with the value `us-west-1`. We will use the flag `--data-urlencode` in curl so it encodes the query parameter for us, in conjunction with the `-G` flag to it appends the data to the URL.

Reviewer: At the end of the line, should it read "flag so it appends the data" instead of "flag to it appends the data"?

> For example, the following request filters the response to only include resources that have a label entry `region` with the value `us-west-1`. We will use the flag `--data-urlencode` in curl so it encodes the query parameter for us, in conjunction with the `-G` flag to it appends the data to the URL.

Reviewer 2: ```suggestion
For example, the following request filters the response to only include resources that have a label entry `region` with the value `us-west-1`. We will use the flag `--data-urlencode` in curl so it encodes the query parameter for us, in conjunction with the `-G` flag so it appends the data to the URL.
```

| Entity | `entity.name` `entity.namespace` `entity.deregister` `entity.entity_class` `entity.subscriptions` |
| Event | `event.name` `event.namespace` `event.check.handlers` `event.check.publish` `event.check.round_robin` `event.check.runtime_assets` `event.check.status` `event.check.subscriptions` `event.entity.deregister` `event.entity.entity_class` `event.entity.subscriptions` |
| Extension | `extension.name` `extension.namespace` |
| Filter | `filter.name` `filter.namespace` `filter.action` `filter.runtime_assets` |
| Handler | `handler.name` `handler.namespace` `handler.filters` `handler.handlers` `handler.mutator` `handler.type`| 
| Hook | `hook.name` `hook.namespace` |
| Mutator | `mutator.name` `mutator.namespace` `mutator.runtime_assets` |
| Namespace | `namespace.name` |
| Role | `role.name` `role.namespace` |
| RoleBinding | `rolebinding.name` `rolebinding.namespace` `rolebinding.role_ref.name` `rolebinding.role_ref.type`|
| Silenced | `silenced.name` `silenced.namespace` `silenced.check` `silenced.creator` `silenced.expire_on_resolve` `silenced.subscription` |
| User | `user.username` `user.disabled` `user.groups` |

#### Supported operators

> There's two _equality-based_ operators supported, `==` (equality) and `!=` (inequality). For example, the following statements are possible:

Reviewer: Should it be "There are" instead of "There is"?

> There's two _equality-based_ operators supported, `==` (equality) and `!=` (inequality). For example, the following statements are possible:

Reviewer 2: ```suggestion
There are two _equality-based_ operators supported, `==` (equality) and `!=` (inequality). For example, the following statements are possible:
```

> There's two _equality-based_ operators supported, `==` (equality) and `!=` (inequality). For example, the following requirements are possible:

Reviewer 2: I think the term "requirements" is a bit confusing. Is there another term we could use? Maybe "statements"?

{{< highlight shell >}}
check.publish == true
check.namespace != "default"
{{< /highlight >}}

> Additionally, there's two _set-based_ operators to deal with list of values, `in` and `notin`. For example, the following statements are possible:

Reviewer: s/there's/there are

> Additionally, there's two _set-based_ operators to deal with list of values, `in` and `notin`. For example, the following statements are possible:

Reviewer 2: ```suggestion
Additionally, there are two _set-based_ operators to deal with lists of values, `in` and `notin`. For example, the following statements are possible:
```

> Additionnaly, there's two _set-based_ operators to deal with list of values, `in` and `notin`. For example, the following requirements are possible:

Reviewer 2: ```suggestion
Additionally, there's two _set-based_ operators to deal with list of values, `in` and `notin`. For example, the following requirements are possible:
```

{{< highlight shell >}}
linux in check.subscriptions
slack notin check.handlers
check.subscriptions in [system,linux]
{{< /highlight >}}

#### Combining selectors and requirements

> A field or label selector can be made of multiple requirements which are separated with the logical operator `&&` (_AND_). For example, the following curl request would look up checks that are configured to be publish **and** have the slack handler:

Reviewer 2: ```suggestion
A field or label selector can be made of multiple requirements which are separated with the logical operator `&&` (_AND_). For example, the following curl request looks up checks that are configured to be published **and** have the `slack` handler:
```
